sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasExec.scala

```diff
@@ -91,6 +93,7 @@ case class TransformWithStateInPandasExec(
   private val sessionLocalTimeZone = conf.sessionLocalTimeZone
   private val pythonRunnerConf = ArrowPythonRunner.getPythonRunnerConfMap(conf)
   private[this] val jobArtifactUUID = JobArtifactSet.getCurrentJobArtifactState.map(_.uuid)
+  private val (dedupAttributes, argOffsets) = resolveArgOffsets(child.output, groupingAttributes)
```

Comment:
Hmm - what does this do internally ? could we add some comment maybe ?

Reply:
This is unchanged in the base change. It was originally here: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasExec.scala#L318. I am only moving this out as a class variable so that other function will need to access it as well.
This function is mainly used to resolve the offsets for restoring key and value attributes in the python side of runner here: https://github.com/apache/spark/blob/master/python/pyspark/worker.py#L1706